railties/lib/rails/engine.rb

```diff
@@ -601,7 +601,7 @@ def load_seed
       routing_paths = paths["config/routes.rb"].existent
       external_paths = self.paths["config/routes"].paths
       routes.draw_paths.concat(external_paths)
-      app.routes.draw_paths.concat(external_paths)
+      app.routes.draw_paths.concat(external_paths) unless app == self
```

Comment:
Honestly, I am not sure this is the best fix.

Reply:
Alternative: we could change `draw_paths` to be initialized as `Set.new` rather than `[]`. But Set does not have a `concat` method. Maybe this might break applications and engines ?